code/modules/clothing/modular/modular.dm

```diff
@@ -0,0 +1,430 @@
+/**
+	Modular armor
+
+	Modular armor consists of a a suit and helmet.
+	The suit is able to have a storage, module, and 3x armor attachments (chest, arms, and legs)
+	Helmets only have a single module slot.
+
+	Suits have a single action, which is to toggle the flashlight.
+	Helmets have diffrnet actions based on what module you have installed.
+
+*/
+/obj/item/clothing/suit/modular
+	name = "Jaeger XM-02 combat exoskeleton"
+	desc = "Designed to mount a variety of modular armor components and support systems. It comes installed with light-plating and a shoulder lamp. Mount armor pieces to it by clicking on the frame with the components"
+	icon = 'icons/mob/modular/modular_armor.dmi'
+	icon_state = "underarmor_icon"
+	item_state = "underarmor"
+	flags_armor_protection = CHEST|GROIN|ARMS|LEGS|FEET|HANDS
+	/// What is allowed to be equipped in suit storage
+	allowed = list(
+		/obj/item/weapon/gun,
+		/obj/item/storage/belt/sparepouch,
+		/obj/item/storage/large_holster/machete,
+		/obj/item/weapon/claymore,
+		/obj/item/storage/belt/gun
+	)
+	flags_equip_slot = ITEM_SLOT_OCLOTHING
+	w_class = WEIGHT_CLASS_BULKY
+
```

Comment:
I don't know how to do the fancy suggestions thing but the /suit/modular lacks a `time_to_unequip` and `time_to_equip` otherwise you can hot equip without delay.